### PR TITLE
16-letter Move Names

### PR DESCRIPTION
**Pull request by Greenphx9**
## Description
Adds a define (B_EXPANDED_MOVE_NAMES) that changes the move name length to 16 characters, and fixes up "squashed" move names (eg: DublIronBash -> Double Iron Bash). Requires the user to make some changes to the summary screen, bag menu, and in-battle move selector to get fully working.

## **Discord contact info**
Greenphx#5428

### CODE REVIEW

File: include/constants/battle_config.h [hunk 1]
```diff
@@ -275,4 +275,6 @@
 
 #define B_ENABLE_DEBUG                  TRUE     // If set to TRUE, enables a debug menu to use in battles by pressing the Select button.
 
+#define B_EXPANDED_MOVE_NAMES           FALSE    // If TRUE, move names are increased from 12 characters to 16 characters.
```
Comment (LOuroboros) [inline include/constants/battle_config.h:278]: Should be put in the "Interface settings" section, below `B_HIDE_HEALTHBOX_IN_ANIMS`.

File: include/constants/global.h [hunk 2]
```diff
@@ -64,6 +64,8 @@
 #define SAVED_TRENDS_COUNT 5
 #define PYRAMID_BAG_ITEMS_COUNT 10
 
+#include "battle_config.h"
```
Comment (LOuroboros) [inline include/constants/global.h:67]:
```suggestion
#include "constants/battle_config.h"
```
There isn't a `include/battle_config.h` file in any of the 3 branches.
Also, this should be put at the top of the file, below `#define GUARD_CONSTANTS_GLOBAL_H` imo in order to remain consistent with the huge majority of the files in the project, kinda like this:
```
#define GUARD_CONSTANTS_GLOBAL_H

#include "constants/battle_config.h"

// Invalid Versions show as "----------" in Gen 4 and Gen 5's summary screen.
```

File: src/data/text/move_names.h [hunk 3]
```diff
@@ -1,3 +1,766 @@
+#if B_EXPANDED_MOVE_NAMES == TRUE
+//16 letters
```
Comment (LOuroboros) [inline src/data/text/move_names.h:2]:
```suggestion
// 16 letters
```

File: src/data/text/move_names.h [hunk 4]
```diff
@@ -1,3 +1,766 @@
...
+    [MOVE_DRAGON_ENERGY] = _("Dragon Energy"),
+    [MOVE_FREEZING_GLARE] = _("Freezing Glare"),
+    [MOVE_FIERY_WRATH] = _("Fiery Wrath"),
+    [MOVE_THUNDEROUS_KICK] = _("Thunderous Kick"),
+    [MOVE_GLACIAL_LANCE] = _("Glacial Lance"),
+    [MOVE_ASTRAL_BARRAGE] = _("Astral Barrage"),
+    [MOVE_EERIE_SPELL] = _("Eerie Spell"),
+};
+#else
+//12 letters
```
Comment (LOuroboros) [inline src/data/text/move_names.h:763]:
```suggestion
// 12 letters
```

File: src/data/text/move_names.h [hunk 5]
```diff
@@ -1,3 +1,766 @@
...
+    [MOVE_BUZZY_BUZZ] = _("Buzzy Buzz"),
+    [MOVE_SIZZLY_SLIDE] = _("Sizzly Slide"),
+    [MOVE_GLITZY_GLOW] = _("Glitzy Glow"),
+    [MOVE_BADDY_BAD] = _("Baddy Bad"),
+    [MOVE_SAPPY_SEED] = _("Sappy Seed"),
+    [MOVE_FREEZY_FROST] = _("Freezy Frost"),
+    [MOVE_SPARKLY_SWIRL] = _("Sparkly Swirl"),
+    [MOVE_VEEVEE_VOLLEY] = _("Veevee Volley"),
+    [MOVE_DOUBLE_IRON_BASH] = _("Double Iron Bash"),
+    //GEN 8
```
Comment (LOuroboros) [inline src/data/text/move_names.h:695]:
```suggestion
    // GEN 8
```
Though I don't know why is this comment here at all. There's no point to it, unless we added one to separate each generation's set of moves.

**Review by LOuroboros — CHANGES_REQUESTED** [review]

File: include/constants/battle_config.h [hunk 6]
```diff
@@ -242,6 +242,7 @@
 #define B_SHOW_TARGETS              TRUE  // If set to TRUE, all available targets, for moves hitting 2 or 3 Pokémon, will be shown before selecting a move.
 #define B_SHOW_SPLIT_ICON           TRUE  // If set to TRUE, it will show an icon in the summary showing the move's category split.
 #define B_HIDE_HEALTHBOX_IN_ANIMS   TRUE  // If set to TRUE, hides healthboxes during move animations.
+#define B_EXPANDED_MOVE_NAMES       FALSE // If TRUE, move names are increased from 12 characters to 16 characters.
```
Comment (LOuroboros) [inline include/constants/battle_config.h:245]:
```suggestion
#define B_EXPANDED_MOVE_NAMES       FALSE // If set to TRUE, move names are increased from 12 characters to 16 characters.
```

**Review by LOuroboros — CHANGES_REQUESTED** [review]

**Review by ghoulslash — DISMISSED** [review]

**Comment by ghoulslash** [comment]
fix conflicts

**Review by ghoulslash — APPROVED** [review]

File: src/data/text/move_names.h [hunk 7]
```diff
@@ -1,3 +1,765 @@
...
+    [MOVE_PLAY_ROUGH] = _("Play Rough"),
+    [MOVE_FAIRY_WIND] = _("Fairy Wind"),
+    [MOVE_MOONBLAST] = _("Moonblast"),
+    [MOVE_BOOMBURST] = _("Boomburst"),
+    [MOVE_FAIRY_LOCK] = _("Fairy Lock"),
+    [MOVE_KINGS_SHIELD] = _("King's Shield"),
+    [MOVE_PLAY_NICE] = _("Play Nice"),
+    [MOVE_CONFIDE] = _("Confide"),
+    [MOVE_DIAMOND_STORM] = _("Diamond Storm"),
+    [MOVE_STEAM_ERUPTION] = _("Steam Erruption"),
```
Comment (sphericalice) [inline src/data/text/move_names.h:597]:
```suggestion
    [MOVE_STEAM_ERUPTION] = _("Steam Eruption"),
```

**Pull request merged**